### PR TITLE
Add codepen button to examples - part 1!

### PR DESCRIPTION
**Pull request by spectranaut**
This PR adds codepen buttons to the following examples! To test locally, you will have to serve the example file with a web server ([described here](https://github.com/w3c/aria-practices/wiki/How-to-add-the-Open-In-Codepen-button-to-an-example-page#how-to-test-your-changes-locally)).

### RawGitHack preview links

- [carousel/carousel-1-prev-next.html](https://raw.githack.com/w3c/aria-practices/codepen-implementation-part1/examples/carousel/carousel-1-prev-next.html)
- [carousel/carousel-2-tablist.html](https://raw.githack.com/w3c/aria-practices/codepen-implementation-part1/examples/carousel/carousel-2-tablist.html)
- [combobox/combobox-select-only.html](https://raw.githack.com/w3c/aria-practices/codepen-implementation-part1/examples/combobox/combobox-select-only.html)
- [menubar/menubar-navigation.html](https://raw.githack.com/w3c/aria-practices/codepen-implementation-part1/examples/menubar/menubar-navigation.html)
- [listbox/listbox-grouped.html](https://raw.githack.com/w3c/aria-practices/codepen-implementation-part1/examples/listbox/listbox-grouped.html)
- [combobox/combobox-datepicker.html](https://raw.githack.com/w3c/aria-practices/codepen-implementation-part1/examples/combobox/combobox-datepicker.html)

### CODE REVIEW

**Comment by jongund** [comment]
@spectranaut 
@mcking65 
For the carousel example is in code pen, if you select one of the links in the carousel it loads the APG page, not just the example code.  Seems like a change in context of the , but may not be a problem for developers look at the code.   Actually in the carousel examples when the page is fully loaded, the user can interact with the configuration options.  Might be a plus?

**Comment by jongund** [comment]
@mcking65 
@spectranaut 
One issue is that the examples rely on the default fonts and maybe some other CSS from the APG example template, so the fonts are noticeably different in the code pen examples.  Maybe need to include some basic CSS for at least the default font so the examples seem consistent between APG and Code Pen.

Otherwise the Code Pen button focus styling looks good and so far all the example load.  The only artifact is that is the example code contains links, activating them in code pen loads the entire APG page.

**Comment by a11ydoer** [comment]
@spectranaut @mcking65 
This is awesome!!!

@jongund 
I agree with your second suggestion. Btw, in your first comment, what "configuration option" are you referring to? I could not find out a configuration option which can solve the issue you are pointing out. Only configuration option I changed in CodePen was layout view, code on the right and datepicker output on the left because the date picker example was had to interact when I had the code view in the upper pane and the date picker in the down pane.

**Comment by ZoeBijl** [comment]
@spectranaut would it be possible to include an `@import` to the base.css file? This way we wouldn’t miss any styling for the components that rely on it :)

**Comment by spectranaut** [comment]
I implement @ZoeBijl's idea to include the default example page styling in `css_external`, so you can see the file is included from the settings page of the codepen but you can't see it in the "css" edit box.

I think it's not such a big problem that clicking it pulls in the whole example page -- unless you really think it is, @jongund ?

**Comment by mcking65** [comment]
@jongund @carmacleod @smhigley @charmarkk @a11ydoer @jnurthen 

I'd like a couple of approving reviews on this. No need to look at any of the code; just make sure that the button on the example page and result in codepen sufficiently meet expectations.

I'd like to merge by Friday if possible.

**Comment by a11ydoer** [comment]
@mcking65 

I have checked 6 examples above. I can confirme that the button on the example page, results in codepen sufficiently meet expectations. 

One side note, one of whistles and bells we can ignore.  It would be nice to transfer the APG example page title to codepen title sincec codepen title is "untitled" for each example.

This is fancy and great help for developers, @spectranaut and @mcking65

**Review by jongund — APPROVED** [review]

File: examples/js/examples.js [hunk 1]
```diff
@@ -401,6 +401,10 @@ function addOpenInCodePenForm(
     css: '',
     js: '',
     head: '<base href="' + location.href + '">',
+    css_external:
```
Comment (ZoeBijl) [inline examples/js/examples.js:404]: Woohoo!

**Review by ZoeBijl — APPROVED** [review]
Nice work :)

**Review by carmacleod — APPROVED** [review]
+1

This is so cool! Thanks, @spectranaut !

Approving as-is, but if you wanted to add a title for @a11ydoer, then I think we could just use:
```
  var path = location.pathname.split('/');
  var title = path.length > 1 ? path[path.length - 2] : ''; // example directory
```

and then just add the following line to the postJson:
```
  title: title,
```

**Pull request merged**